modules/kafka/src/main/java/com/devonfw/module/kafka/common/messaging/api/config/MessageSenderConfig.java

```diff
@@ -0,0 +1,181 @@
+  public ProducerFactory<Object, Object> messageKafkaProducerFactory(KafkaCommonProperties messageKafkaCommonProperties,
+      KafkaProducerProperties messageKafkaProducerProperties) {
+
+    return createProducerFactory(messageKafkaCommonProperties, messageKafkaProducerProperties);
+  }
+
+  /**
+   * Creates bean for {@link KafkaTemplate}.
+   *
+   *
+   * @param messageProducerLoggingListener the {@link #messageProducerLoggingListener(MessageLoggingSupport, Tracer)}
+   * @param messageKafkaProducerFactory the
+   *        {@link #messageKafkaProducerFactory(KafkaCommonProperties, KafkaProducerProperties)}
+   *
+   * @return the KafkaTemplate
+   */
+  @Bean
+  public KafkaTemplate<Object, Object> messageKafkaTemplate(
+      ProducerLoggingListener<Object, Object> messageProducerLoggingListener,
+      ProducerFactory<Object, Object> messageKafkaProducerFactory) {
+
+    return createKafkaTemplate(messageProducerLoggingListener, messageKafkaProducerFactory);
+  }
+
+  /**
+   * Creates bean for {@link MessageSenderImpl}.
+   *
+   *
+   * @param messageKafkaTemplate the {@link #messageKafkaTemplate(ProducerLoggingListener, ProducerFactory)}
+   * @param messageLoggingSupport the {@link MessageCommonConfig#messageLoggingSupport()}
+   * @param messageSenderProperties the {@link #messageSenderProperties()}
+   * @param diagnosticContextFacade the {@link DiagnosticContextFacade}
+   * @param messageSpanInjector the {@link MessageSpanInjector}
+   * @param tracer the {@link Tracer}
+   * @return the MessageSenderImpl.
+   */
+  @Bean
+  public MessageSenderImpl messageSender(KafkaTemplate<Object, Object> messageKafkaTemplate,
+      MessageLoggingSupport messageLoggingSupport, MessageSenderProperties messageSenderProperties,
+      DiagnosticContextFacade diagnosticContextFacade, MessageSpanInjector messageSpanInjector) {
+
+    MessageSenderImpl bean = new MessageSenderImpl();
+    bean.setKafkaTemplate(messageKafkaTemplate);
+    bean.setLoggingSupport(messageLoggingSupport);
+    bean.setSenderProperties(messageSenderProperties);
+    bean.setSpanInjector(messageSpanInjector);
+    bean.setTracer(this.tracer);
+    bean.setDiagnosticContextFacade(diagnosticContextFacade);
+    return bean;
+  }
+
+  /**
+   * Creates bean for {@link ProducerLoggingListener}.
+   *
+   * @param messageLoggingSupport {@link MessageLoggingSupport}
+   * @param tracer {@link Tracer}
+   * @return the ProducerLoggingListener
+   */
+  @Bean
+  public ProducerLoggingListener<Object, Object> messageProducerLoggingListener(
+      MessageLoggingSupport messageLoggingSupport) {
+
+    return new ProducerLoggingListener<>(messageLoggingSupport, this.tracer);
+  }
+
+  /**
+   * This method is used to create {@link KafkaTemplate}.
+   *
+   *
+   * @param producerLogListener the {@link ProducerLoggingListener}
+   * @param producerFactory the {@link ProducerFactory}
+   * @return KafkaTemplate
+   */
+  public static KafkaTemplate<Object, Object> createKafkaTemplate(
```

Comment:
Why is this method static instead of providing the template as bean?

Reply:
yes i agree this doesnt require to be static but its an unnecessary of keeping it as bean . it is a callable method which creates a bean for KafkaTemplate in the same class